packages/dataprovider/src/buildQuery.ts

```diff
@@ -65,6 +65,11 @@ export const buildQueryFactory = (
       params,
     )!;
 
+    if (otherOptions.queryDialect === "typegraphql") {
+      Object.keys(variables).forEach((key) => {
+        variables[key] = variables[key] || undefined;
+      });
+    }
```

Comment:
careful here:

- `variables[key] || undefined;` will convert falsy values to undefined. That is probably not what you want. (falsy values are e.g. 0, "", null, undefined, false).
- `variables[key] ?? undefined;` i slightly better as it only converts null to undefined. But that also may break certain use cases. E.g. check this issue: https://github.com/panter/ra-data-prisma/issues/48

Instead, only convert exactly these params that fail in the nest/prisma graphql dialect.
Or: use the introspection result to check which params are not allowed to be null. 

If its only orderBy, then check the function `buildOrderBy`, it returns null by default. Check what happens with this param. I guess you could also just return `undefined` there.

Reply:
Good catch, I did not know ?? operator